SparkKit/BasicButton.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  Copyright © 2017 Spark App Studio. All rights reserved.
+//
+
+import UIKit
+
+@IBDesignable
+class BasicButton: UIButton {
+
+    let rounded: CGFloat = 12
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        sharedInit()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        sharedInit()
+    }
+
+    func sharedInit() {
+        titleLabel?.textColor = UIColor.white
+    }
+
+
+    override func draw(_ rect: CGRect) {
+
+        let ctx = UIGraphicsGetCurrentContext()!
+
+        let inset = rect.insetBy(dx: 1, dy: 1)
+        let rounded = UIBezierPath(roundedRect: inset, cornerRadius: self.rounded)
+
+        UIColor.blue.setFill()
+        ctx.addPath(rounded.cgPath)
+        ctx.fillPath()
+
+        UIColor.orange.setStroke()
+        ctx.addPath(rounded.cgPath)
+        ctx.strokePath()
+    }
+
+    override var intrinsicContentSize: CGSize {
+        if let string = titleLabel?.text as
+            NSString? {
+            let stringSize = string.size(withAttributes: [.font: titleLabel!.font])
+            return CGSize(width: rounded * 2 + ceil(stringSize.width), height: rounded * 2 + ceil(stringSize.height))
+        }
+        return super.intrinsicContentSize
+    }
+
+    override func prepareForInterfaceBuilder() {
+        self.setNeedsDisplay()
```

Comment:
this is more of a question from me. will this help interface builder to display IBDesignable? my interface builder for many time struggles to to display IBDesignables properly..

Reply:
I'm not totally sure if this is needed... I'll try without before I merge it. 
But I know you can override this to do other stuff for your designables..... like populate some default data and stuff.

>When Interface Builder instantiates a class with the IB_DESIGNABLE attribute, it calls this method to let the resulting object know that it was created at design time. You can implement this method in your designable classes and use it to configure their design-time appearance. For example, you might use the method to configure a custom text control with a default string. The system does not call this method; only Interface Builder calls it.

also the docs say...
>Your implementation of this method must call super at some point so that parent classes can perform their own custom setup.
OOPS